NuRadioMC/SignalProp/analyticraytracing.py

```diff
@@ -1744,7 +1749,362 @@ def get_path(self, iS, n_points=1000):
+        direction: 3-dim array of floats
+            the propagation direction (must be normalized to 1)
+        nx: float
+            the index of refraction into the x direction
+        ny: float
+            the index of refraction into the y direction
+        nz: float
+            the index of refraciton into the z direction
+
+        
+        """
+        sx = direction[0]
+        sy = direction[1]
+        sz = direction[2]
+        A = (nx**2 - n**2)*(ny**2 - n**2)*(nz**2 - n**2)
+        B = n**2 *( sx**2 *(ny**2 - n**2)*(nz**2 - n**2) + sy**2 *(nx**2 - n**2)*(nz**2 - n**2) + sz**2 *(nx**2 - n**2)*(ny**2 - n**2))
+
+        return A + B
+
+    def get_polarization(self, n, direction, nx, ny, nz, rounding=20):
+        
+        """
+        returns the polarization of a wave for the direction of propagation 
+    
+        Parameters
+        ------------
+        n: float
+            the index of effective refraction into the x direction calculated by get_effective_index        
+        direction: numpy.array
+            propagation direction of the wave              
+        nx: float
+            the index of refraction into the x direction        
+        ny: float
+            the index of refraction into the y direction        
+        nz: float
+            the index of refraciton into the z direction         
+        rounding: int
+            the precision of the comparison between n and nx, ny, nz for the special cases    
+        """ 
+        
+        if(np.round(n, rounding) in [np.round(nx, rounding), np.round(ny, rounding), np.round(nz, rounding)]):
+        # treat special cases here
+            if(np.isclose(direction[0], 1)):  # propagation into x
+                print(f"propagating into x")
+                if(np.isclose(n, ny)):
+                    return np.array([0, 1, 0])
+                elif(np.isclose(n, nz)):
+                    return np.array([0, 0, 1])
+            elif(np.isclose(direction[1], 1)):  # propagation into y
+                print(f"propagating into y")
+                if(np.isclose(n, nx)):
+                    return np.array([1, 0, 0])
+                elif(np.isclose(n, nz)):
+                    return np.array([0, 0, 1])
+            elif(np.isclose(direction[2], 1)):  # propagation into z
+                print(f"propagating into z")
+                if(np.isclose(n, nx)):
+                    return np.array([1, 0, 0])
+                elif(np.isclose(n, ny)):
+                    return np.array([0, 1, 0])
+        # now treat the cases where the direction is perpendicular to just one of the principal axes
+            elif(np.isclose(np.dot(direction, np.array([1, 0, 0])), 0)):  # direction perpendicular to x-axis
+                if(np.isclose(n, nx)):
+                    return np.array([1, 0, 0])
+                else:
+                    print(f"ERROR: propagation direction perpendicular to x-axis but n not equal to n_x {n}")
```

Comment:
Please use the logger as it is done everywhere else. Then the user can specify which level of warning or Error they want to see.

Reply:
This is for ALL print statements. Print statements should NOT be used in any module.

Reply:
I have not commented in detail on all other print statements. Please fix all of them.